src/main/java/com/work/management/service/employee/impl/EmployeeServiceImpl.java

```diff
@@ -95,7 +97,13 @@ public EmployeeDto updateEmployeeEntity(EmployeeDto employeeDto) {
 
   @Override
   @Transactional(isolation = Isolation.SERIALIZABLE, propagation = Propagation.REQUIRED)
-  public List<Employee> bulkUpdate(BulkEmployeeDto bulkEmployeeDto) {
+  public ImmutableList<Employee> bulkUpdate(BulkEmployeeDto bulkEmployeeDto) throws Exception {
+    if (!hasUniqueIds(bulkEmployeeDto.getEmployeeIds())) {
+      ExceptionUtils
+          .throwBadRequestException(
+              String.format("Duplicate employee ids exist"));
```

Comment:
remove String.format....its not needed here, you are not building a dynamic string here, its only needed for building dynamic string.